### PR TITLE
[Form Recognizer] Troubleshooting guide 

### PR DESCRIPTION
**Pull request by samvaity**

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+You may have seen a Build model error, mostly when you are trying to build your custom model.
+It looks like this:
+
+```text
+com.azure.core.exception.HttpResponseException: Invalid model created with model Id [modelId], errorCode: [2012], message: <specific-reason>"
+```
+
+#### Invalid training data set
+This error indicates that the provided data set does not match the training data requirements.
+Learn more about building a training data set, [here](https://docs.microsoft.com/en-us/azure/applied-ai-services/form-recognizer/how-to-guides/build-custom-model-v3)
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md:54]: update to api link here.

File: sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+The exception looks like this:
+```text
+com.azure.core.exception.HttpResponseException: Invalid request, errorCode: [ModelBuildError], message: Could not build the model: OCR file 'Form_1.jpg.ocr.json' has an invalid schema"
+```
+
+#### Invalid SAS URL
+You may have seen this error like:
+```
+com.azure.core.exception.HttpResponseException: Invalid model created with model Id [modelId], errorCode: [2012], message: Unable to list blobs on the Azure blob storage account.
+```
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md:77]: The other error sections put the example error below the description, this one should follow that pattern too?

File: sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+com.azure.core.exception.HttpResponseException: Status code 200, Invalid model created with model Id [modelId], errorCode: [3014], message: Generic error during training."
+```
+
+### Unexpected time to build a custom model
+It is common to notice a longer time than what is expected to build a custom model when using `DocumentBuildMode: "neural"`.
+Depending on the service load you can usually expect it to ~10min.
+
+For simpler use-cases, you can use [Custom Template models] which are easy-to-train models and take lesser time (in seconds) to build over the 
+[Custom Neural models] that are deep learned models.
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md:94]: There seem to be some links missing here

File: sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+### Enable HTTP request/response logging
+
+Reviewing the HTTP request sent or response received over the wire to/from the Azure Form Recognizer service can be useful in
+troubleshooting issues. To enable logging the HTTP request and response payload, the `ContainerRegistryClient` can be configured as shown below:
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md:99]:
```suggestion
Reviewing the HTTP request sent or response received over the wire to/from the Azure Form Recognizer service can be useful in
troubleshooting issues. To enable logging the HTTP request and response payload, the `DocumentAnalysisClient` can be configured as shown below:
```

File: sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+       * [Invalid SAS URL](#invalid-sas-url)
+    * [Generic Error](#generic-error)
+* [Unexpected time to build a custom model](#unexpected-time-to-build-a-custom-model)
+* [Enable HTTP request/response logging](#enable-http-requestresponse-logging)
+
+## Troubleshooting Errors & Exceptions
+### Handling HttpResponseException
+Form Recognizer service methods throw a [HttpResponseException](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/exception/HttpResponseException.java) or its subclass on failure.
+The HttpResponseException thrown by the Azure Form Recognizer client library includes detailed response error object
+that provides specific useful insights into what went wrong and include corrective actions to fix common issues.
```
Comment (kinelski) [inline sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md:19]: nit: "and include" => "and includes"

**Review by kinelski — COMMENTED** [review]
In .NET we had a couple of customers (around 3~4) incorrectly trying to use the old `FormRecognizerClient` class (which uses V2) with the model they trained in FR Studio (which uses V3). I wonder if it's worth checking the error messages we get in this case and including it in the guide.

In particular, a common exception they got in .NET was `FormatException`, thrown because they tried passing a custom model ID that's not a GUID (allowed in V3, not allowed in V2).

File: sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+administrationAsyncClient.deleteModel("{modelId}")
+    .doOnSuccess(
+        ignored -> System.out.println("Success!"))
+    .doOnError(
+        error -> error instanceof ResourceNotFoundException,
+        error -> System.out.println("Exception: Delete could not be performed."));
+```
+
+### Build model error
+You may have seen a Build model error, mostly when you are trying to build your custom model.
```
[the remark this answers is not in the document]
Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md:45]: Yes, that is what the next section is about. ##### Invalid training data set

**Comment by samvaity** [comment]
> `erClient` class (which uses V2) with the model they trained in FR Studio (which uses V3). I wonder if it's worth checking the error messages we get in this case and including it in the guide.
> 
> In particular, a common exception they got in .NET was `FormatException`, thrown because they tried passing a custom model ID that's not a GUID (allowed in V3, not allowed in V2).

We throw an IllegalArgumentException in this case saying:
Exception in thread "main" java.lang.IllegalArgumentException: Invalid UUID string: <modelID>

Maybe we could check for the API version and accordingly update the error message to specify the user to target the newer endpoint.
Thoughts @kinelski, @catalinaperalta ?

File: sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md:45]: Ah I see it now. Thanks

**Comment by kinelski** [comment]
> > `erClient` class (which uses V2) with the model they trained in FR Studio (which uses V3). I wonder if it's worth checking the error messages we get in this case and including it in the guide.
> > In particular, a common exception they got in .NET was `FormatException`, thrown because they tried passing a custom model ID that's not a GUID (allowed in V3, not allowed in V2).
> 
> We throw an IllegalArgumentException in this case saying: Exception in thread "main" java.lang.IllegalArgumentException: Invalid UUID string: `<modelID>`
> 
> Maybe we could check for the API version and accordingly update the error message to specify the user to target the newer endpoint. Thoughts @kinelski, @catalinaperalta ?

That would be ideal. However, in .NET if a customer is trying to use the old `FormRecognizerClient`, there's no way they could be targeting V3 (our design doesn't allow it). Because of this we can't really tell if they're trying to use V3 but picked the wrong client, or if they actually want to use V2 but passed an invalid model name.

If you haven't got any negative feedback about it in Java, I think it's fair to leave it alone for now. I'll probably need to add a section about it in the troubleshooting guide in .NET, though.

**Review by kinelski — APPROVED** [review]
LGTM!

**Comment by catalinaperalta** [comment]
> > `erClient` class (which uses V2) with the model they trained in FR Studio (which uses V3). I wonder if it's worth checking the error messages we get in this case and including it in the guide.
> > In particular, a common exception they got in .NET was `FormatException`, thrown because they tried passing a custom model ID that's not a GUID (allowed in V3, not allowed in V2).
> 
> We throw an IllegalArgumentException in this case saying: Exception in thread "main" java.lang.IllegalArgumentException: Invalid UUID string: `<modelID>`
> 
> Maybe we could check for the API version and accordingly update the error message to specify the user to target the newer endpoint. Thoughts @kinelski, @catalinaperalta ?

We havent seen this error much, but I will check what our error response looks like and see about mentioning it in the troubleshooting guide, I think it may be a common enough scenario particularly since we hope customers will be migrating from v2 to v3.

**Review by catalinaperalta — APPROVED** [review]

**Pull request merged**